langstream-cli/src/main/java/ai/langstream/cli/commands/docker/LocalRunApplicationCmd.java

```diff
@@ -281,12 +282,10 @@ private void executeOnDocker(
             boolean startDatabase,
             boolean dryRun)
             throws Exception {
-        File tmpInstanceFile = Files.createTempFile("instance", ".yaml").toFile();
-        Files.write(tmpInstanceFile.toPath(), instanceContents.getBytes(StandardCharsets.UTF_8));
+        File tmpInstanceFile = createReadableTempFile("instance", instanceContents);
         File tmpSecretsFile = null;
         if (secretsContents != null) {
-            tmpSecretsFile = Files.createTempFile("secrets", ".yaml").toFile();
-            Files.write(tmpSecretsFile.toPath(), secretsContents.getBytes(StandardCharsets.UTF_8));
+            tmpSecretsFile = createReadableTempFile("secrets", secretsContents);
```

Comment:
if this file is containing secrets then it should be readable only by the current user